libsledge/src/numeric_instructions.c

```diff
@@ -54,60 +54,48 @@ rotr_u64(uint64_t n, uint64_t c_u64)
 INLINE uint32_t
 u32_div(uint32_t a, uint32_t b)
 {
-	if (unlikely(b == 0)) sledge_abi__wasm_trap_raise(WASM_TRAP_ILLEGAL_ARITHMETIC_OPERATION);
 	return a / b;
 }
 
 INLINE uint32_t
 u32_rem(uint32_t a, uint32_t b)
 {
-	if (unlikely(b == 0)) sledge_abi__wasm_trap_raise(WASM_TRAP_ILLEGAL_ARITHMETIC_OPERATION);
 	return a % b;
 }
 
 INLINE int32_t
 i32_div(int32_t a, int32_t b)
 {
-	if (unlikely(b == 0 || (a == INT32_MIN && b == -1)))
-		sledge_abi__wasm_trap_raise(WASM_TRAP_ILLEGAL_ARITHMETIC_OPERATION);
 	return a / b;
 }
 
 INLINE int32_t
 i32_rem(int32_t a, int32_t b)
 {
-	if (unlikely(b == 0 || (a == INT32_MIN && b == -1)))
-		sledge_abi__wasm_trap_raise(WASM_TRAP_ILLEGAL_ARITHMETIC_OPERATION);
 	return a % b;
 }
 
 INLINE uint64_t
 u64_div(uint64_t a, uint64_t b)
 {
-	if (unlikely(b == 0)) sledge_abi__wasm_trap_raise(WASM_TRAP_ILLEGAL_ARITHMETIC_OPERATION);
 	return a / b;
 }
 
 INLINE uint64_t
 u64_rem(uint64_t a, uint64_t b)
 {
-	if (unlikely(b == 0)) sledge_abi__wasm_trap_raise(WASM_TRAP_ILLEGAL_ARITHMETIC_OPERATION);
 	return a % b;
 }
 
 INLINE int64_t
 i64_div(int64_t a, int64_t b)
 {
-	if (unlikely(b == 0 || (a == INT64_MIN && b == -1)))
```

Comment:
I don't understand the second condition here. If `a == INT64_MIN && b == -1` will this always cause a signal?

Reply:
I believe the second condition is an overflow because under two's complement, the negative side can represent one more number than the positive side.

Reply:
I changed the `traps` app so that the numerator was INT32_MIN to test this, and it seemed to trap. I temporarily made this a long to confirm that longs behave the same way. On Intel at least, both seem to trap with a SIGFPE. I guess that means it's not just floating-point units that trigger this.

Reply:
Awesome!

Reply:
yeah, I think that sigfpe includes div by zero, which is applicable for integers as well.